docs/reduction_toolbox.rst

```diff
@@ -428,6 +432,112 @@ which should be removed rather than propagated:
     plt.show()
 
 
+Working with multi-extension FITS image files
+---------------------------------------------
+
+Multi-extension FITS (MEF) image files cannot be processed natively in ``ccdproc``. The example below illustrates how to `~ccdproc.flat_correct` all of the extensions in a MEF and write out the calibrated file as a MEF. Applying other reduction steps would be similar.
+
+The example begins by creating small MEF "science" image and a sample flat:
+
+    >>> import numpy as np
+    >>>
+    >>> from astropy.utils.misc import NumpyRNGContext
+    >>> from astropy.io import fits
+    >>> from astropy.nddata import CCDData
+    >>>
+    >>> from ccdproc import flat_correct
+    >>>
+    >>>
+    >>> def make_sample_mef(science_name, flat_name):
+    ...     """
+    ...     Make a multi-extension FITS image with random data
+    ...     and a MEF flat.
+    ...
+    ...     Parameters
+    ...     ----------
+    ...
+    ...     science_name : str
+    ...         Name of the science image created by this function.
+    ...
+    ...     flat_name : str
+    ...         Name of the flat image created by this function.
+    ...     """
+    ...     with NumpyRNGContext(1234):
+    ...         number_of_image_extensions = 3
+    ...         science_image = [fits.PrimaryHDU()]
+    ...         flat_image = [fits.PrimaryHDU()]
+    ...         for _ in range(number_of_image_extensions):
+    ...             # Simulate a cloudy night, average pixel
+    ...             # value of 100 with a read_noise of 1 electron.
+    ...             # Image size is 150 × 150.
+    ...             data = np.random.normal(100., 1.0, [150, 150])
```

Comment:
Is 150 really needed? In the interest of the test-runs and so on we could also use 10 x 10, right?

That saves a bit of memory and time.

Reply:
Changed